### PR TITLE
ice_stran: add on_valid_pair() to callbacks

### PR DESCRIPTION
**Pull request by 257**
Hello,
This is related to ```71ff0065108470602af063d631e0affad38efc04```.

### CODE REVIEW

**Comment by sauwming** [comment]
This is related to issue #2263 .

It seems okay, but do you need the parameter op in the `on_valid_pair()` callback? (since it seems that it's always PJ_ICE_STRANS_OP_FIRST_CAND anyway?)

**Comment by 257** [comment]
> This is related to issue #2263 .
> 
> It seems okay, but do you need the parameter op in the `on_valid_pair()` callback? (since it seems that it's always PJ_ICE_STRANS_OP_FIRST_CAND anyway?)

Yeah, sticky. I don't have a perfect solution.

My thinking was that it would help applications that already have an ```on_ice_complete```; they can reuse the same function, possibly modifying it a bit, and react accordingly based on ```op```.

Here is an example of it I made for Jami here:
https://review.jami.net/c/ring-daemon/+/13902/7/src/ice_transport.cpp#513

**Comment by sauwming** [comment]
I would prefer to remove op parameter from the callback since it doesn't seem to be necessary.

As for your application, if you want to use the same callback handler function, I believe you can do like this?
```
#define PJ_ICE_STRANS_OP_FIRST_CAND [some_value]

tr->onComplete(ice_st, PJ_ICE_STRANS_OP_FIRST_CAND, PJ_SUCCESS);
```

**Comment by 257** [comment]
> I would prefer to remove op parameter from the callback since it doesn't seem to be necessary.
> 
sure, updated in a rush; no mistakes, hopefully :)

btw, code style is diverging more and more. what do you think of a .clang-format? i can work on it if need be.

File: pjnath/include/pjnath/ice_strans.h [hunk 1]
```diff
@@ -181,6 +181,15 @@ typedef struct pj_ice_strans_cb
     void    (*on_data_sent)(pj_ice_strans *sock,
 			    pj_ssize_t sent);
 
+    /**
+     * An optional callback that will be called by the ICE transport when a
+     * valid pair has been found during ICE negotiation.
+     *
+     * @param ice_st	    The ICE stream transport.
+     * @param status	    Operation status.
```
Comment (sauwming) [inline pjnath/include/pjnath/ice_strans.h:189]: remove this non-existent param

**Comment by sauwming** [comment]
Not sure what you're referring to, but the coding style seems fine?

The patch looks okay, only a minor fix needed. Will also need to wait for others to review it as well.

File: pjnath/include/pjnath/ice_strans.h (hunk 1 above)
Comment (257) [reply pjnath/include/pjnath/ice_strans.h]: oops! fixed.

**Pull request merged**